app/src/main/java/com/maazm7d/termuxhub/data/source/local/LocalDataSource.kt

```
@@ -0,0 +1,56 @@
+    override fun getFavoritesFlow(): Flow<List<ToolEntity>> = toolDao.getFavoritesFlow()
+    override suspend fun getToolById(id: String): ToolEntity? = toolDao.getToolById(id)
+    override suspend fun insertTools(tools: List<ToolEntity>) = toolDao.insertAll(tools)
+    override suspend fun updateTool(tool: ToolEntity) = toolDao.update(tool)
+
+    override suspend fun loadMetadataFromAssets(fileName: String): MetadataDto? = withContext(Dispatchers.IO) {
+        try {
+            val input = appContext.assets.open(fileName)
+            val text = BufferedReader(InputStreamReader(input)).use { it.readText() }
+            val adapter = moshi.adapter(MetadataDto::class.java)
+            adapter.fromJson(text)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)